src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -230,7 +230,7 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 		}));
 
 		this._register(this._connectionManagementService.onDisconnect(async (connectionParams) => {
-			if (this.isObjectExplorerConnectionUri(connectionParams.connectionUri)) {
+			if (this.isObjectExplorerConnectionUri(connectionParams.connectionUri) && connectionParams?.connectionProfile) {
```

Comment:
If connectionParams is undefined here it'll throw on `connectionParams.connectionUri`

Did you mean to just do `connectionParams.connectionProfile`? Although if connectionProfile is actually possibly undefined that type should be updated too...